### PR TITLE
Home task for playwright actions

### PR DESCRIPTION
**Pull request by Podigor**

### CODE REVIEW

File: pages/homePage/components/SignUpPopup.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    this.getNameWrongLengthMessage = this._nameWrongLengthMessage
+    this.getLastNameRequiredMessage = this._lastNameRequiredMessage
+    this.getLastNameInvalidMessage = this._lastNameInvalidMessage
+    this.getLastNameWrongLengthMessage = this._lastNameWrongLengthMessage
+    this.getEmailRequiredMessage = this._emailRequiredMessage
+    this.getEmailInvalidMessage = this._emailInvalidMessage
+    this.getPasswordRequiredMessage = this._passwordRequiredMessage
+    this.getPasswordWrongLengthMessage = this._passwordWrongLengthMessage
+    this.getRepeatPasswordRequiredMessage = this._repeatPasswordRequiredMessage
+    this.getRepeatPasswordDoNotMatchMessage = this._repeatPasswordDoNotMatchMessage
```
Comment (sttaran) [inline pages/homePage/components/SignUpPopup.js]: what is it? 

it is not getters

File: tests/user-registration.spec.js [hunk 2]
```diff
@@ -0,0 +1,504 @@
...
+test.describe("Successful user registration", ()=> {
+
+    test.beforeEach(async ({page})=> {
+        const homePage = new HomePage(page)
+        const signInPopup = new SignInPopup(page)
+        const signUpPopup = new SignUpPopup(page)
+
+        await homePage.visitHomePage()
+        await homePage.openSignInPopup()
+        await signInPopup.openRegistrationPopup()
```
Comment (sttaran) [inline tests/user-registration.spec.js]: we have "sign up" button right on the home page

File: pages/homePage/components/SignUpPopup.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  constructor(page) {
+    this.page = page
+    this.getSignUpPopup = page.locator('app-signup-modal')
+    this.getNameInput = this.getSignUpPopup.locator('#signupName')
+    this.getLastNameInput = this.getSignUpPopup.locator('#signupLastName')
+    this.getEmailInput = this.getSignUpPopup.locator('#signupEmail')
+    this.getPasswordInput = this.getSignUpPopup.locator('#signupPassword')
+    this.getRepeatPasswordInput = this.getSignUpPopup.locator('#signupRepeatPassword')
+    this.getRegisterBtn = this.getSignUpPopup.getByRole('button', {name: 'Register'})
+    this.getValidationMessage = this.getSignUpPopup.locator('.invalid-feedback').locator('p')
```
Comment (sttaran) [inline pages/homePage/components/SignUpPopup.js]: do not name locator with `get` prefix

File: tests/user-registration.spec.js [hunk 4]
```diff
@@ -0,0 +1,504 @@
...
+            repeatPassword: 'Secret10'
+        }
+        await expect(signUpPopup.getRegisterBtn).toBeDisabled()
+        await signUpPopup.fillRegistrationForm(userData)
+        await expect(signUpPopup.getRegisterBtn).toBeDisabled()
+        await expect(signUpPopup.getNameInput).toHaveClass(/is-invalid/) //checks that border color is red by using corresponding class
+        await expect(signUpPopup.getLastNameInput).not.toHaveClass(/is-invalid/)
+        await expect(signUpPopup.getEmailInput).not.toHaveClass(/is-invalid/)
+        await expect(signUpPopup.getPasswordInput).not.toHaveClass(/is-invalid/)
+        await expect(signUpPopup.getRepeatPasswordInput).not.toHaveClass(/is-invalid/)
```
Comment (sttaran) [inline tests/user-registration.spec.js]: also check that input with invalid value has red border

**Review by sttaran — CHANGES_REQUESTED** [review]

File: pages/homePage/components/SignUpPopup.js (hunk 1 above)
Comment (Podigor) [reply pages/homePage/components/SignUpPopup.js]: fixed

File: tests/user-registration.spec.js (hunk 2 above)
Comment (Podigor) [reply tests/user-registration.spec.js]: fixed

File: pages/homePage/components/SignUpPopup.js (hunk 3 above)
Comment (Podigor) [reply pages/homePage/components/SignUpPopup.js]: fixed

File: tests/user-registration.spec.js (hunk 4 above)
Comment (Podigor) [reply tests/user-registration.spec.js]: fixed

**Review by sttaran — APPROVED** [review]

**Pull request merged**